espnet2/tasks/asr.py

```diff
@@ -189,6 +190,7 @@
         whisper=OpenAIWhisperDecoder,
         hugging_face_transformers=HuggingFaceTransformersDecoder,
         s4=S4Decoder,
+        linear_decoder=LinearDecoder,
```

Comment:
It is not a good indication to add a classification-oriented decoder here. I personally would vote for a new task, but indeed it would introduce a range of workload.

If the current style is our ultimate option, I would suggest to add more warning/info msg when using the decoder and to have a compatibility check of the decoder and the data during forward

Reply:
Noted, I will update with a TODO to move it to cls1 task. For now can we check this version in?

Reply:
Added here https://github.com/espnet/espnet/pull/5977/commits/3749554dcbc89a6417995bb0e73b8503a538c714
ptal.

Reply:
Sounds super! Then, I feel it would be a good start~ Thanks for the clarification.